src/core/tsi/s2a/handshaker/s2a_handshaker_client.cc

```diff
@@ -289,8 +305,49 @@ void s2a_handshaker_client_destroy(S2AHandshakerClient* client) {
 
 /** ------------------- Testing methods. -------------------------- **/
 
-grpc_byte_buffer* S2AHandshakerClient::get_send_buffer_for_testing() {
-  return send_buffer_;
+void S2AHandshakerClient::set_grpc_caller_for_testing(s2a_grpc_caller caller) {
+  if (is_test_) {
+    grpc_caller_ = caller;
+  }
+}
+
+grpc_metadata_array* S2AHandshakerClient::initial_metadata_for_testing() {
```

Comment:
The logic here is a little confusing. I expected that if `is_test_` is true, `nullptr` should be returned which does not seem to be the case.

Reply:
Since this is a `for_testing` method, it should only return something interesting if `is_test_` is true. If `is_test_` is false, then it should return `nullptr`. (The confusion here might be related to the previous comment, about the comments on the `for_testing` functions that were unclear.)

Reply:
I see. Please also add above comments in the code to make it clear.

Reply:
Clarified this comment in the next commit.